### PR TITLE
Add Storage configuration section to KVM compose form.

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Added Storage configuration section to KVM compose form.
- Updated TagSelector so it can be properly styled without a label
- Rich select to be added in https://github.com/canonical-web-and-design/maas-squad/issues/2061

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the KVM list and select a LXD KVM
- Open the compose dropdown and check that there is a "Storage configuration" section.
- Check that hovering over the "Add disk" button shows a tooltip explaining that you can only have one disk for LXD KVMs.
- Change "Size" to below 8 and check that a warning shows.
- Change "Size" to above what's available in the storage pool and check that en error shows.
- Enter a valid size and click "Compose machine" and check that a new machine shows up in the machine list with the correct storage.
- Go to the KVM details page of a virsh KVM and check that you can add multiple disks with different storage pools.

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2043

## Screenshot
0 0 0 0_8400_MAAS_r_kvm_190

### CODE REVIEW

**Comment by squidsoup** [comment]
The spacing feels quite tight here:

image

File: ui/src/app/kvm/components/KVMActionFormWrapper/ComposeForm/ComposeForm.test.js [hunk 1]
```diff
@@ -106,9 +110,17 @@ describe("ComposeForm", () => {
   });
 
   it("can handle composing a machine", () => {
```
Comment (squidsoup) [inline ui/src/app/kvm/components/KVMActionFormWrapper/ComposeForm/ComposeForm.test.js:112]: how about `it("can compose a machine"...`

**Comment by squidsoup** [comment]
A minor ux bug here, if you delete the value in Size (with backspace), the field resets to `0`, however if you enter a new value, you get `0[new value]`:

image

File: ui/src/app/store/pod/types.ts [hunk 2]
```diff
@@ -16,8 +16,9 @@ export type PodHintExtras = {
   local_disks: number;
 };
 
-export type PodStoragePool = Model & {
+export type PodStoragePool = {
   available: number;
+  id: string;
```
Comment (squidsoup) [inline ui/src/app/store/pod/types.ts:21]: 😢

File: ui/src/app/kvm/components/KVMActionFormWrapper/ComposeForm/ComposeForm.tsx [hunk 3]
```diff
@@ -157,6 +220,45 @@ const ComposeForm = ({ setSelectedAction }: Props): JSX.Element | null => {
...
         .min(1, "Cores must be a positive number.")
         .max(available.cores, `Only ${available.cores} cores available.`),
+      disks: Yup.array().of(
+        Yup.object().shape({
+          id: Yup.number().required("ID is required"),
+          location: Yup.string().required("Location is required"),
+          size: Yup.number("Size is required")
+            .min(1, "At least 1GB required")
+            .required("Size is required")
+            .test("enoughSpace", "Not enough space", function test() {
```
[the remark this answers is not in the document]
Comment (Caleb-Ellis) [reply ui/src/app/kvm/components/KVMActionFormWrapper/ComposeForm/ComposeForm.tsx]: Yup!

File: ui/src/app/store/pod/types.ts (hunk 2 above)
Comment (Caleb-Ellis) [reply ui/src/app/store/pod/types.ts:21]: I know...

**Comment by Caleb-Ellis** [comment]
> It feels like the spacing here is quite tight, does that look right to you?
> 
> image

Updated padding to be 0.25rem larger on the top and bottom:
0 0 0 0_8400_MAAS_r_kvm_190 (2)

**Pull request merged**